rescorer.go

```diff
@@ -0,0 +1,116 @@
+//  Copyright (c) 2025 Couchbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bleve
+
+import (
+	"github.com/blevesearch/bleve/v2/fusion"
+	"github.com/blevesearch/bleve/v2/search/query"
+)
+
+// Rescorer is applied after all the query and knn results are obtained.
+// The main use of Rescorer is in hybrid search; all the individual scores
+// for query and knn are combined using Rescorer
+type rescorer interface {
```

Comment:
i think i'm missing somehting, but can you explain why an interface is necessary over here? do you see there being a different implementation altogether of these methods in the future? because the functions defined and implemented look pretty straightforward and generic for multiple rescoring strategies. 

also i think defining an interface at the bleve level (which has a lot of code); although it does hide it from the user, it doesn't necessarily hide the different implementations (thereby making things safe) within the codebase itself since the non-scoring related constructs is also exposed to this interface and thereby making it a bit more vulnerable.